mlflow/tracking/fluent.py

```diff
@@ -3049,3 +3050,74 @@ def setup_autologging(module):
             register_post_import_hook(setup_autologging, "pyspark", overwrite=True)
         if "pyspark.ml" in target_library_and_module:
             register_post_import_hook(setup_autologging, "pyspark.ml", overwrite=True)
+
+
+_ACTIVE_MODEL_ID = ContextVar("active_model_id", default=None)
+
+
+class ActiveModel(LoggedModel):
```

Comment:
do we need this class? Can we decorate `set_active_model` with `conetxtlib.context_manager`?

Reply:
If we decorate it with @contextmanager it can only use as context manager?

Reply:
oh you're right. Never mind this comment.

Reply:
Can we make `LoggedModel` a context manager?

Reply:
Emm we could, but we do not want to always set active_model_id when a LoggedModel is created?

Reply:
Can we set the active id in `__enter__`?

Reply:
No because if `set_active_model` is not used as a context manager, then `__enter__` is not triggered.